src/engine/renderer/tr_shader.cpp

```diff
@@ -5992,209 +5990,188 @@ A second parameter will cause it to print in sorted order
+	shaderSortStrings[shaderSort_t::SS_SEE_THROUGH] = "SEE_THROUGH";
+	shaderSortStrings[shaderSort_t::SS_BANNER] = "BANNER";
+	shaderSortStrings[shaderSort_t::SS_FOG] = "FOG";
+	shaderSortStrings[shaderSort_t::SS_UNDERWATER] = "UNDERWATER";
+	shaderSortStrings[shaderSort_t::SS_WATER] = "WATER";
+	shaderSortStrings[shaderSort_t::SS_FAR] = "FAR";
+	shaderSortStrings[shaderSort_t::SS_MEDIUM] = "MEDIUM";
+	shaderSortStrings[shaderSort_t::SS_CLOSE] = "CLOSE";
+	shaderSortStrings[shaderSort_t::SS_BLEND0] = "BLEND0";
+	shaderSortStrings[shaderSort_t::SS_BLEND1] = "BLEND1";
+	shaderSortStrings[shaderSort_t::SS_BLEND2] = "BLEND2";
+	shaderSortStrings[shaderSort_t::SS_BLEND3] = "BLEND3";
+	shaderSortStrings[shaderSort_t::SS_BLEND6] = "BLEND6";
+	shaderSortStrings[shaderSort_t::SS_ALMOST_NEAREST] = "ALMOST_NEAREST";
+	shaderSortStrings[shaderSort_t::SS_NEAREST] = "NEAREST";
+	shaderSortStrings[shaderSort_t::SS_POST_PROCESS] = "POST_PROCESS";
+
+	std::string shaderSortHeader = "shaderSort";
+	int maxShaderSortStringLen = shaderSortHeader.length();
+
+	for ( shaderSortStringMap::iterator iter = shaderSortStrings.begin();
+		iter !=shaderSortStrings.end(); iter++ )
+	{
+		maxShaderSortStringLen = std::max( maxShaderSortStringLen, (int)iter->second.length() );
+	}
+
+	using stageTypeStringMap = std::unordered_map<stageType_t, std::string>;
+	stageTypeStringMap stageTypeStrings;
+	stageTypeStrings[stageType_t::ST_COLORMAP] = "COLORMAP";
+	stageTypeStrings[stageType_t::ST_GLOWMAP] = "GLOWMAP";
+	stageTypeStrings[stageType_t::ST_DIFFUSEMAP] = "DIFFUSEMAP";
+	stageTypeStrings[stageType_t::ST_NORMALMAP] = "NORMALMAP";
+	stageTypeStrings[stageType_t::ST_HEIGHTMAP] = "HEIGHTMAP";
+	stageTypeStrings[stageType_t::ST_PHYSICALMAP] = "PHYSICALMAP";
+	stageTypeStrings[stageType_t::ST_SPECULARMAP] = "SPECULARMAP";
+	stageTypeStrings[stageType_t::ST_REFLECTIONMAP] = "REFLECTIONMAP";
+	stageTypeStrings[stageType_t::ST_REFRACTIONMAP] = "REFRACTIONMAP";
+	stageTypeStrings[stageType_t::ST_DISPERSIONMAP] = "DISPERSIONMAP";
+	stageTypeStrings[stageType_t::ST_SKYBOXMAP] = "SKYBOXMAP";
+	stageTypeStrings[stageType_t::ST_SCREENMAP] = "SCREENMAP";
+	stageTypeStrings[stageType_t::ST_PORTALMAP] = "PORTALMAP";
+	stageTypeStrings[stageType_t::ST_HEATHAZEMAP] = "HEATHAZEMAP";
+	stageTypeStrings[stageType_t::ST_LIQUIDMAP] = "LIQUIDMAP";
+	stageTypeStrings[stageType_t::ST_LIGHTMAP] = "LIGHTMAP";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_lighting_PBR] = "LIGHTING_PBR";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_lighting_PHONG] = "LIGHTING_PHONG";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_reflection_CB] = "REFLECTION_CB";
+	stageTypeStrings[stageType_t::ST_ATTENUATIONMAP_XY] = "ATTENUATIONMAP_XY";
+	stageTypeStrings[stageType_t::ST_ATTENUATIONMAP_Z] = "ATTENUATIONMAP_XZ";
+
+	std::string stageTypeHeader = "stageType";
+	int maxStageTypeStringLen = stageTypeHeader.length();
+
+	for ( stageTypeStringMap::iterator iter = stageTypeStrings.begin();
```

Comment:
Use for-each loops. `for (const auto& kv : stageTypeStrings)` then access with `kv.second`.

Reply:
Done.